File: src/leetcode/majorityelement/java/MajorityElement.java
package leetcode.majorityelement.java;

/*
* Given an array nums of size n, return the majority element.
The majority element is the element that appears more than ⌊n / 2⌋ times. You may assume that the majority element always exists in the array.
* */

import java.util.*;
import java.util.stream.Collectors;

public class MajorityElement {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        Map<Integer,Integer> majorities = new HashMap<>();

        for (int num : nums) {
            if (majorities.containsKey(num)) {
                majorities.put(num, majorities.get(num) + 1);
            } else {
                majorities.put(num, 1);
            }
        }

        var majority = majorities.entrySet().stream().sorted(Map.Entry.comparingByValue()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (oldValue, newValue) -> oldValue, LinkedHashMap::new));

        System.out.println(majority.entrySet());
        int currentMajority = 0;
        for (Map.Entry<Integer, Integer> e : majority.entrySet()){
            currentMajority = e.getKey();
        }

        System.out.println(currentMajority);
        return currentMajority;
    }
}

File: src/leetcode/majorityelement/kotlin/MajorityElement.kt
package leetcode.majorityelement.kotlin

class MajorityElement {
    fun majorityElement(nums: IntArray): Int {
        nums.sort()
        val majorities = mutableMapOf<Int, Int>()

        for (num in nums) {
            if (num in majorities) {
                majorities[num] = majorities[num]!! + 1
            } else {
                majorities[num] = 1
            }
        }

        val majority = majorities.entries.sortedBy { it.value }.associate { it.toPair() }
        println(majority.entries)
        var currentMajority = 0
        for (e in majority.entries) {
            currentMajority = e.key
        }

        println(currentMajority)
        return currentMajority
    }
}